### PR TITLE
feat: dynamically fetch clarity grammar file for editor

### LINKED ISSUE #926: use clarity grammar and docs from clarity repo in code editor
https://github.com/hirosystems/clarity-repl/blob/1282a1e2d5a6d759e0df923f4bb30a13b27ea124/src/clarity/docs/mod.rs
https://github.com/hirosystems/clarinet/blob/develop/components/clarity-vscode/syntaxes/clarity.tmLanguage.json

### PR DESCRIPTION
**Pull request by leahjlou**
closes #926

### CODE REVIEW

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -31,7 +31,7 @@ interface ExplorerAppProps extends AppProps {
 
 function ExplorerApp({ Component, ...rest }: ExplorerAppProps): React.ReactElement {
   const { apiServer, networkMode, pageProps } = rest;
-  const { isHome, fullWidth } = pageProps;
+  const { isHome, fullWidth, ...restPageProps } = pageProps;
```
Comment (leahjlou) [inline src/pages/_app.tsx:34]: The other page props needed to be forwarded to the component for `getStaticProps` to work

File: src/pages/_app.tsx [hunk 2]
```diff
@@ -75,7 +80,7 @@ const handleNetworkModeQueryParam = (store: EnhancedStore, appContext: AppContex
   console.log('[debug] activeNetwork', activeNetwork);
   console.log('[debug] networks', networks);
   const queryNetworkMode = ((Array.isArray(query.chain) ? query.chain[0] : query.chain) ||
-    '') as NetworkModes;
+    NetworkModes.Mainnet) as NetworkModes;
```
Comment (leahjlou) [inline src/pages/_app.tsx:83]: I've never seen an error with this in prod, but locally I was encountering issues with this being an empty string and the network mode being set to `undefined`. Is Mainnet a reasonable default here?

Comment (He1DAr) [reply src/pages/_app.tsx:83]: yep! I fixes this in my latest PR

**Review by He1DAr — APPROVED** [review]
This is great!

**Pull request merged**